MessageProcessing/AnomalyDetection/AnomalyDetector.cs

```diff
@@ -0,0 +1,70 @@
+﻿using MessageProcessing;
+using MessageProcessing.AnomalyDetection;
+using MessageProcessing.AnomalyDetection.Enum;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+
+public class AnomalyDetector : IAnomalyDetector
+{
+    private readonly AnomalyDetectionConfig _anomalyDetectionConfig;
+
+    public AnomalyDetector(
+        IOptions<AnomalyDetectionConfig> anomalyDetectionConfig)
+    {
+        _anomalyDetectionConfig = anomalyDetectionConfig.Value;
+    }
+
+    public List<AnomalyType> DetectAnomalies(ServerStatistics currentStatistics, ServerStatistics previousStatistics)
+    {
+        List<AnomalyType> anomalies = new();
+
+        if (IsMemoryUsageAnomaly(currentStatistics.MemoryUsage, previousStatistics.MemoryUsage))
+        {
+            anomalies.Add(AnomalyType.MemoryUsageAnomaly);
+        }
+
+        if (IsCpuUsageAnomaly(currentStatistics.CpuUsage, previousStatistics.CpuUsage))
+        {
+            anomalies.Add(AnomalyType.CPUUsageAnomaly);
+        }
+
+        if (IsHighMemoryUsage(currentStatistics))
+        {
+            anomalies.Add(AnomalyType.HighMemoryUsage);
+        }
+
+        if (IsHighCpuUsage(currentStatistics.CpuUsage))
+        {
+            anomalies.Add(AnomalyType.HighCPUUsage);
+        }
+        return anomalies;
+    }
+
+    public bool IsMemoryUsageAnomaly(double currentMemoryUsage, double previousMemoryUsage)
```

Comment:
let's think how we can implement this class in a better way and more extendable way.